notebooks/daymet_access.ipynb

```diff
@@ -0,0 +1,916 @@
+      "text/plain": [
+       "<Figure size 640x480 with 2 Axes>"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "prcp_7day_sum = chena_huc8['prcp'].rolling(time=7, center=False, min_periods=1).sum()\n",
+    "prcp_7day_sum.sel(time='2000-06-21').plot(vmin=0)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Speaking of aggregation, let's do just that! For each daily time step, lets average the 7-day cumulative rainfall over the entire watershed. We will write this to a new variable in the `chena_huc8` dataset."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "chena_huc8['basin_mean_7d_prcp'] = prcp_7day_sum.groupby('time').mean(dim=[\"x\",\"y\"])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Using a similar process to the DAYMET OPeNDAP requests, we can request some USGS stream gauge data to add context to our Chena River basin precipitation statistics. We will get daily mean streamflow data for the year 2000 from the USGS Chena River gauge location by using the USGS water services API. (The gauging station details can be found [here](https://waterdata.usgs.gov/nwis/inventory/?site_no=15514000&agency_cd=USGS) and more info on the API can be found [here](https://waterservices.usgs.gov/docs/statistics/statistics-details/).)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 12,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "gauge_id = '15514000'\n",
+    "start_dt = '2000-05-01'\n",
+    "end_dt = '2000-10-31'\n",
+    "gauge_url = \"https://waterservices.usgs.gov/nwis/dv/?format=json&sites=\" + gauge_id + \"&startDT=\" + start_dt + \"&endDT=\" + end_dt + \"&siteStatus=all\"\n",
+    "\n",
+    "with requests.Session() as s:\n",
+    "    response = s.get(gauge_url).json()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Parse the JSON response to get gauge location coordinates (we will use these in a later plot), and create a dictionary of observation dates and stream gauge measurements. While parsing, convert the date string into a numpy datetime object and drop the hours/minutes/seconds details; this will allow a 1:1 comparison to timesteps in our precipitation data.\n",
+    "\n",
+    "For each time step in the precipitation dataset, query the dictionary to find streamflow data for that date and populate a list with the results. If no data is available for that date, populate with NAN print a message. Convert the list to an array and add this to our precipitation dataset as a new variable, creating our final result dataset for plotting."
```

Comment:
Is the "print a message" text relevant here? I don't see any print statements.

Reply:
There is a print statement in the "except" block, which will print a message if no gauge data is found. In this example, we don't hit that error but I thought I would include it for anyone copy/pasting from this notebook into their own workflow. Catching errors at the request seems to be a good practice.